ios/RNWatson.swift

```diff
@@ -40,21 +41,32 @@ class RNSpeechToText: RCTEventEmitter {
   }
 
   @objc func startStreaming(_ languageCustomizationID: String?, errorCallback: @escaping RCTResponseSenderBlock) {
+    self.isListening = true
+    self.accumulator = SpeechRecognitionResultsAccumulator()
+    
     var settings = RecognitionSettings(contentType: "audio/ogg")
     settings.interimResults = true
     settings.smartFormatting = true
+
     var callback = RecognizeCallback()
 
-    callback.onResults = { results  in
+    callback.onResults = { payload in
       if(self.hasListeners)
       {
-        self.accumulator.add(results: results)
-        self.sendEvent(withName: "StreamingText", body: self.accumulator.bestTranscript)
+        self.accumulator.add(results: payload)
+        let isFinal = payload.results?.last?.finalResults ?? true
```

Comment:
Has JavaScript become Swift? 🤔

Reply:
BTW, @breadadams  shouldn't that `?? true` be `?? false`. Shouldn't you expect that if there are no results `isFinal` must be false?

Reply:
Ok I've been messing around with this @gabrielperales, comparing the web and iOS results - because on web I have the fallback as `false` and iOS it's `true`...

So, I set it to `true` on iOS because during trials a *few* times I got an empty `results` array. If you press the stop button in-app after this you get no more data back, and the app gets stuck in "waiting mode".

This _never_ happens on the web app, the websocket always delivers a response for me, and I've not been able to recreate the issue of an empty array there.

This means we should be safe to switch it to `true` on both platforms, because the flag is always there on web from what I've seen (so our `true` never really gets used there).

The other option would be to set it to `false` on both platforms and add some form of timeout listener in the frontend, so it never gets stuck in waiting mode for X seconds. That may be overthinking it though.